File: CS/Human.cs
namespace StudentManagementSystem;

public abstract class Human
{
    private string _name;
    private short _age;

    public Human(string name, short age)
    {
        _name = name;
        _age = age;
    }

    public string Name => _name;
    public short Age => _age;
}

File: Rust/src/human.rs
pub trait Human {
	fn get_name(&self) -> &str;
	fn get_age(&self) -> i8;
}

#[macro_export]
macro_rules! human_new {
	{} => {
		pub fn new(name: &str, age: i8) -> Self {
			Self { name: name.into(), age }
		}
	}
}

#[macro_export]
macro_rules! human_getters {
	{} => {
		fn get_name(&self) -> &str {
			&self.name
		}
		fn get_age(&self) -> i8 {
			self.age
		}
	}
}
